policy/modules/admin/usbguard.te

```diff
@@ -0,0 +1,58 @@
+policy_module(usbguard, 1.0)
+
+########################################
+#
+# Declarations
+#
+
+## <desc>
+##	<p>
+##	Determine whether usbguard-daemon can modify its rules in
+##	/etc/usbguard.
+##	</p>
+## </desc>
+gen_tunable(usbguard_modify_rule_files, false)
+
+type usbguard_t;
+type usbguard_daemon_exec_t;
+init_daemon_domain(usbguard_t, usbguard_daemon_exec_t)
+
+type usbguard_etc_t;
+files_config_file(usbguard_etc_t)
+
+# /dev/shm
+type usbguard_tmpfs_t;
+files_tmpfs_file(usbguard_tmpfs_t)
+
+type usbguard_var_log_t;
+logging_log_file(usbguard_var_log_t)
+
+########################################
+#
+# Usbguard local policy
+#
+
+allow usbguard_t self:capability { chown dac_override dac_read_search fowner };
+allow usbguard_t self:netlink_kobject_uevent_socket create_socket_perms;
+allow usbguard_t self:unix_stream_socket rw_stream_socket_perms;
+
+dev_rw_sysfs(usbguard_t)
+
+files_read_etc_files(usbguard_t)
+list_dirs_pattern(usbguard_t, usbguard_etc_t, usbguard_etc_t)
+read_files_pattern(usbguard_t, usbguard_etc_t, usbguard_etc_t)
+tunable_policy(`usbguard_modify_rule_files',`
+	manage_dirs_pattern(usbguard_t, usbguard_etc_t, usbguard_etc_t)
+	manage_files_pattern(usbguard_t, usbguard_etc_t, usbguard_etc_t)
+')
+
+manage_dirs_pattern(usbguard_t, usbguard_tmpfs_t, usbguard_tmpfs_t)
+manage_files_pattern(usbguard_t, usbguard_tmpfs_t, usbguard_tmpfs_t)
+mmap_read_files_pattern(usbguard_t, usbguard_tmpfs_t, usbguard_tmpfs_t)
+fs_tmpfs_filetrans(usbguard_t, usbguard_tmpfs_t, { dir file })
+
+append_files_pattern(usbguard_t, usbguard_var_log_t, usbguard_var_log_t)
+create_files_pattern(usbguard_t, usbguard_var_log_t, usbguard_var_log_t)
+logging_log_filetrans(usbguard_t, usbguard_var_log_t, file)
+manage_files_pattern(usbguard_t, usbguard_var_log_t, usbguard_var_log_t)
```

Comment:
You might be able to do without this. not sure.. however either you use the finer grained append_files_pattern + create_files_pattern + ??? or you use manage_files_pattern. no point in using both

Reply:
ok